src/components/organisms/ProfileModal/UserInformationContent/UserInformationContent.tsx

```diff
@@ -23,47 +27,51 @@ const UserInformationContent: React.FunctionComponent<PropsType> = ({
   setIsPasswordEditMode,
   hideModal,
 }) => {
-  const { user, profile } = useUser();
+  const { user, profile, userWithId } = useUser();
   const profileQuestions = useSelector(
     (state) => currentVenueSelectorData(state)?.profile_questions
   );
   const venueId = useVenueId();
+  const venue = useSelector(currentVenueSelector);
 
   const history = useHistory();
   const firebase = useFirebase();
-  const logout = () => {
-    firebase.auth().signOut();
+
+  const logout = useCallback(async () => {
+    await firebase.auth().signOut();
+
     // we need to hide the modal because if we already are on the Entrance Page, history.push has no effect
     hideModal();
     history.push(IS_BURN ? "/enter" : venueId ? venueLandingUrl(venueId) : "/");
-  };
+  }, [firebase, hideModal, history, venueId]);
 
-  if (!user) return <></>;
-
-  const toggleKidsMode = () => {
-    if (profile) {
+  const toggleKidsMode = useCallback(async () => {
+    if (user && profile) {
       profile.kidsMode = !profile?.kidsMode;
-      updateUserProfile(user.uid, { kidsMode: profile.kidsMode });
+      await updateUserProfile(user.uid, { kidsMode: profile.kidsMode });
```

Comment:
OOS/future - good catch, this would've silently failed / hit and error boundary. Wonder if there's a lint rule for this.

Reply:
Backlogged in https://github.com/sparkletown/internal-sparkle-issues/issues/91